Documentation Updates (#1868)
PennyLaneAI/pennylane: doc/introduction/interfaces/numpy.rst

It can now be used like any other Python/NumPy function:

>>> phi = np.array([0.5, 0.1], requires_grad=True)

> >>> theta = np.array(0.2, requires_grad=True)

Reviewer: How come these were changed?

Reviewer 2: @antalszava I think you are right it is not necessary as we don't use `qml.grad` after it.

> >>> phi = np.array([0.5, 0.1], requires_grad=True)
> >>> theta = np.array(0.2, requires_grad=True)

Author: ```suggestion
>>> phi = np.array([0.5, 0.1])
>>> theta = 0.2
```

.. code-block:: python

    phi = np.array([0.5, 0.1], requires_grad=True)

>     theta = np.array(0.2, requires_grad=True)

Reviewer: Wouldn't the output on line 118 change too?

Author: I get

(array([-0.47942554,  0.        ]), array(-6.9388939e-18))
which is essentially the same thing.

For example, you may want to pass arguments to a QNode but *not* have
PennyLane consider them when computing gradients.

> Currently, all positional arguments provided to the QNode are assumed to be differentiable
> by default, but this behavior is undergoing deprecation. In a future release, only arguments explicitly
> marked as trainable will be treated as trainable. Not explicitly marking arguments currently raises a 
> deprecation warning.

Reviewer: ```suggestion
All positional arguments provided to the QNode are assumed to be differentiable
by default, but this behaviour is deprecated. In a future release, only arguments explicitly
marked as trainable or marked using the `argnum` keyword argument will be treated as trainable. Not explicitly marking arguments currently raises a 
deprecation warning.
```

> For now, arguments are internally turned into arrays of the PennyLane NumPy module,

Reviewer: ```suggestion
For now, arguments are internally turned into arrays from the PennyLane NumPy module,
```

For ``data``, which is a PennyLane NumPy array, we can simply specify ``requires_grad=False``:

> >>> rng = np.random.default_rng(seed=42)

Reviewer: ```suggestion
>>> rng = np.random.default_rng(seed=42)  # make the results reproducable
```

If we were to naively try computing the gradient of ``circuit5`` using the :func:`~.grad` function,

>>> g1 = qml.grad(circuit5)
>>> params = np.array([np.pi/2, 0.2], requires_grad=True)
>>> g1(params)

> TypeError: Grad only applies to real scalar-output functions. Try jacobian, elementwise_grad or holomorphic_grad.

Reviewer: Nice to include the message! :100:

    def cost(x):
        return np.abs(circuit(x) - 0.5) ** 2

>     params = np.array([0.011, 0.012, 0.05], requires_grad=True)

Reviewer: Did the output after line 381 not change?

Author: OK.... 
```
>>> minimize(cost, params, method='BFGS', jac=qml.grad(cost))
```

is raising a warning... How would we fix that?

Author: fixed.
